presto-main/src/main/java/com/facebook/presto/tdigest/Centroid.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to Ted Dunning under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tdigest;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A single centroid which represents a number of data points.
+ */
+public class Centroid implements Comparable<Centroid>, Serializable {
+    private static final AtomicInteger uniqueCount = new AtomicInteger(1);
+
+    private double centroid = 0;
+    private int count = 0;
+
+    // The ID is transient because it must be unique within a given JVM. A new
+    // ID should be generated from uniqueCount when a Centroid is deserialized.
+    private transient int id;
+
+    private List<Double> actualData = null;
```

Comment:
any reason why not use `DoubleList` in fast util? A list of packaged Double class is quite memory intensive, also these small objects might increase GC overhead?